src/physlr-molecules.cc

```diff
@@ -323,6 +329,78 @@ biconnectedComponents(graph_t& subgraph, vertexToComponent_t& vertexToComponent)
 		}
 		++moleculeNum;
 	}
+	return moleculeNum;
+}
+
+void
+bin_components(
+    componentToVertexSet_t& source,
+    componentToVertexSet_t& binned_neighbours,
+    uint64_t bin_size = 50)
+{
+	// //   Iterate over each component and if its bigger than bin_size:
+	// //   randomly split the component (set of vertices) into smaller even bins
+
+	std::vector<uint64_t> components_size;
+	uint64_t neighborhood_size;
+	uint64_t components_count;
+	for (uint64_t i = 0; i < source.size(); i++) { // NOLINT
+		neighborhood_size = source[i].size();
+		components_count = ((neighborhood_size - 1) / bin_size) + 1;
+		components_size.push_back(components_count);
+	}
+	uint64_t new_size =
+	    std::accumulate(components_size.begin(), components_size.end(), uint64_t(0));
+	binned_neighbours.resize(new_size);
+	uint64_t counter_new = 0;
+	uint64_t base_com_size;
+	uint64_t leftover;
+
+	for (uint64_t i = 0; i < source.size(); i++) { // NOLINT
+		// Using unordered_set, we make use of its random nature and we do not shuffle randomly
+		base_com_size = source[i].size() / components_size[i];
+		leftover = source[i].size() % components_size[i];
+		uint64_t yet_leftover = (leftover ? 1 : 0);
+
+		auto elementIt = source[i].begin();
+		while (elementIt != source[i].end()) {
+			uint64_t length = base_com_size + yet_leftover;
+			if (--leftover == 0) {
+				yet_leftover = 0;
+			}
+
+			for (uint64_t j = 0; j < length; j++) {
+				if (counter_new >= binned_neighbours.size()) {
+					std::cerr << " WAS NOT EXPECTED 1!" << std::endl;
```

Comment:
Are you going to change these error messages to something more descriptive?

Reply:
mmm, I can even remove these comments as this situation is not expected at all. It was for testing when developing. what do you think?

Reply:
I mean if you want to do error checking Im all for it, but if you don't think they will ever occur then you should remove it. If you do keep them, I'd prefer more descriptive error messages.